src/seq/nucleotide.jl

```diff
@@ -97,27 +97,27 @@ end
 
 Test if nucleotide is surely a purine.
 """
-function ispurine(nt::Nucleotide)
+@inline function ispurine(nt::Nucleotide)
     bits = reinterpret(UInt8, nt)
-    return bits != 0 && (bits & 0b1010) == 0
+    return (bits != 0) & ((bits & 0b1010) == 0)
```

Comment:
@bicycle1885 @kdmurray91 

I've just run some benchmarks on my work machine, and I've found actually there's no difference in speed with the change of logical operators, and that both produce simd llvm instructions.

My current explanation for this is - as it states in the julia manual, the compiler can sometimes turn short `&&`, `||`, and `?:` expressions into straight-line code. Which is probably what is happening here. However on my home laptop, simd llvm instructions were not emitted, perhaps the compiler on my home machine didn't turn them into straight line code, and so I didn't get use of simd in the loops.

So ultimately I guess it depends on what we decide. Currently is see no detriment to performance with this change on one laptop, and yet I saw a gain to performance on my home laptop when I wrote this a few days ago. So if this change guarantees simdable code whereas the short circuit operators only sometimes get turned into simdable code, I'm all for this change on the balance of things. But I'm going to repeat the benchmarks I've done on my home machine again, to make absolutely sure.